packages/@aws-cdk/aws-appsync/lib/schema-intermediate.ts

```diff
@@ -259,3 +258,88 @@ export class InputType implements IIntermediateType {
     this.definition[options.fieldName] = options.field;
   }
 }
+
+/**
+ * Properties for configuring an Enum Type
+ *
+ * @param definition - the strings that define this enum type
+ *
+ * @experimental
+ */
+export interface EnumTypeProps {
+  /**
+   * the attributes of this type
+   */
+  readonly definition: string[];
+}
+
+/**
+ * Enum Types are abstract types that includes a set of fields
+ * that represent the strings this type can create.
+ *
+ * @experimental
+ */
+export class EnumType implements IIntermediateType {
+  /**
+   * the name of this type
+   */
+  public readonly name: string;
+  /**
+   * the attributes of this type
+   */
+  public readonly definition: { [key: string]: IField };
```

Comment:
is `definition` the right terminology here? I thought the enum definition is inclusive of both the enum name and the allowed values. however, this would just be the allowed values right?

Reply:
yeah so this is necessary because of the connected use of `IIntermediateType`, since other types (`Input`, `Interface`, and `Object`) need a key and value definition, I just decided to use the key here.

Reply:
most customers wont use the definition because there is little to gain from it, so i think that its fine here

basically the key's of the definition object define the allowed values..

i.e. our definition would look like this

{
  "NEWHOPE": GraphqlType.string(),
  "EMPIRE": GraphqlType.string(),
  "JEDI": GraphqlType.string()
}

Reply:
I see what you mean. where do we expect it to be used? does it need to be exposed?

Reply:
its exposed primarily for homegrown usage..

for example, if we dont have certain functionality, users can still map over the definition and dynamically adjust fields.